asllib/doc/asl.spec

```diff
@@ -1295,10 +1297,3181 @@ typing function annotate_literal(tenv: static_envs, l: literal) -> (t: ty)
+  {width} indicating the number of bits in the
+  bitvector type that contains {field}, in an
+  environment {tenv}, resulting in an annotated bitfield
+  {new_field} or a \typingerrorterm{}, if one is
+  detected.",
+  prose_application = "\hyperlink{relation-annotatebitfield}{annotating} bitfield {field} with width {width} in {tenv} yields {new_field} and {ses}"
+};
+
+relation check_slices_in_width(tenv: static_envs, width: Z, slices: list0(slice)) ->
+         (constants_set(True)) | type_error
+{
+  "checks whether the slices in {slices} fit within the
+  bitvector width given by {width} in {tenv}, yielding
+  $\True$. \ProseOtherwiseTypeError",
+  prose_application = "\hyperlink{relation-checkslicesinwidth}{verifying} slices {slices} fit within width {width} in {tenv} yields True"
+};
+
+relation check_positions_in_width(width: Z, positions: powerset(Z)) ->
+         (constants_set(True)) | type_error
+{
+  "checks whether the set of positions in {positions} fit
+  within the bitvector width given by {width}, yielding
+  $\True$. \ProseOtherwiseTypeError",
+  prose_application = "\hyperlink{relation-checkpositionsinwidth}{verifying} positions {positions} fit within width {width} yields True"
+};
+
+relation disjoint_slices_to_positions(tenv: static_envs, is_static: Bool, slices: list0(slice)) ->
+         (positions: powerset_finite(Z)) | type_error
+{
+  "returns the set of integers defined by the list of
+  slices in {slices} in {positions}. In particular, this
+  rule checks that the following properties:
+  \begin{itemize}
+  \item bitfield slices do not overlap; and
+  \item bitfield slices are not defined in reverse (e.g., \texttt{0:1} rather than \texttt{1:0})
+  \end{itemize}
+  Conducting the checks for
+  these properties requires evaluating the expressions
+  comprising the slices, either via static evaluation of
+  via normalization. The flag {is_static} determines
+  whether the slice is assumed to consist of
+  \staticallyevaluableterm{} expressions. If so, the
+  slice expressions are statically evaluated, and
+  otherwise they are normalized.
+  \ProseOtherwiseTypeError",
+  prose_application = "\hyperlink{relation-disjointslicestopositions}{converting} disjoint slices {slices} in {tenv} with static flag {is_static} yields positions {positions}"
+};
+
+relation bitfield_slice_to_positions(tenv: static_envs, is_static: Bool, slice: slice) ->
+         (positions: option(powerset_finite(Z))) | type_error
+{
+  "returns the set of integers defined by the bitfield
+  slice {slice} in {positions}, if it can be determined
+  via static evaluation or normalization, depending on
+  {is_static}, and $\None$ if it cannot be determined.
+  \ProseOtherwiseTypeError",
+  prose_application = "\hyperlink{relation-bitfieldslicetopositions}{converting} slice {slice} in {tenv} with static flag {is_static} yields optional positions {positions}"
+};
+
+relation eval_slice_expr(tenv: static_envs, is_static: Bool, e: expr) ->
```

Comment:
The layout is strange here: why the parenthesis on the left are smaller than the variable overnames, but on the right they are bigger.

<img width="525" height="90" alt="Image" src="https://github.com/user-attachments/assets/be2be5a6-4584-4dd5-96cd-7267a6406ec2" />